WordPressShared/WordPressShared/Core/Views/WPStyleGuide+DynamicType.swift

```diff
@@ -0,0 +1,182 @@
+        let traits = [UIFontWeightTrait: weight]
+        fontDescriptor = fontDescriptor.addingAttributes([UIFontDescriptorTraitsAttribute: traits])
+        return UIFont(descriptor: fontDescriptor, size: CGFloat(0.0))
+    }
+
+    /// Creates a UIFont for the user current text size settings and calculates its size.
+    ///
+    /// - Parameters:
+    ///     - style: The desired UIFontTextStyle.
+    ///
+    /// - Returns: The created font point size.
+    ///
+    public class func fontSizeForTextStyle(_ style: UIFontTextStyle) -> CGFloat {
+        let fontDescriptor = UIFontDescriptor.preferredFontDescriptor(withTextStyle: style)
+        let font = UIFont(descriptor: fontDescriptor, size: CGFloat(0.0))
+        return font.pointSize
+    }
+
+    /// Creates a NotoSerif UIFont for the user current text size settings.
+    ///
+    /// - Parameters:
+    ///     - style: The desired UIFontTextStyle.
+    ///
+    /// - Returns: The created font.
+    ///
+    public class func notoFontForTextStyle(_ style: UIFontTextStyle) -> UIFont {
+        return self.customNotoFontNamed("NotoSerif", forTextStyle: style)
+    }
+
+    /// Creates a NotoSerif Bold UIFont for the user current text size settings.
+    ///
+    /// - Parameters:
+    ///     - style: The desired UIFontTextStyle.
+    ///
+    /// - Returns: The created font.
+    ///
+    public class func notoBoldFontForTextStyle(_ style: UIFontTextStyle) -> UIFont {
+        return self.customNotoFontNamed("NotoSerif-Bold", forTextStyle: style)
+    }
+
+    /// Creates a NotoSerif Italic UIFont for the user current text size settings.
+    ///
+    /// - Parameters:
+    ///     - style: The desired UIFontTextStyle.
+    ///
+    /// - Returns: The created font.
+    ///
+    public class func notoItalicFontForTextStyle(_ style: UIFontTextStyle) -> UIFont {
+        return self.customNotoFontNamed("NotoSerif-Italic", forTextStyle: style)
+    }
+
+    /// Creates a NotoSerif BoldItalic UIFont for the user current text size settings.
+    ///
+    /// - Parameters:
+    ///     - style: The desired UIFontTextStyle.
+    ///
+    /// - Returns: The created font.
+    ///
+    public class func notoBoldItalicFontForTextStyle(_ style: UIFontTextStyle) -> UIFont {
+        return self.customNotoFontNamed("NotoSerif-BoldItalic", forTextStyle: style)
+    }
+
+    /// Creates a Noto UIFont for the user current text size settings.
+    ///
+    /// - Parameters:
+    ///     - fontName: the Noto font name (NotoSerif, NotoSerif-Bold, NotoSerif-Italic, NotoSerif-BoldItalic)
+    ///     - style: The desired UIFontTextStyle.
+    ///
+    /// - Returns: The created font point size.
+    ///
+    private class func customNotoFontNamed(_ fontName: String, forTextStyle style: UIFontTextStyle) -> UIFont {
+        _ = notoLoaded
+        let fontDescriptor = UIFontDescriptor.preferredFontDescriptor(withTextStyle: style)
+        return UIFont(name: fontName, size: fontDescriptor.pointSize)!
```

Comment:
Noto seems to require views to reload for the size changes to take effect, at least in the reader where I was testing. This may be a limitation of dynamically loaded fonts, but I'll leave a not in case a solution comes to mind.